LeetCodeSolutions/LeafSimilarTrees/LeafSimilarTrees.cs

```diff
@@ -0,0 +1,54 @@
+namespace LeetCodeSolutions;
+public class LeafSimilarTreesSolution
+{
+    private class TreeWrapper(TreeNode node)
+    {
+        private readonly List<int> treeLeaves = new();
+        private readonly TreeNode current = node;
+
+        public void GetValueSequence()
+        {
+            PopulateTreeLeaves(current);
+
+        }
+
+        private void PopulateTreeLeaves(TreeNode node)
+        {
+            if (node.left == null && node.right == null)
+            {
+                treeLeaves.Add(node.val);
+                return;
+            }
+
+            if (node.left != null)
```

Comment:
can move this check to a base case and reduce some lines of code within the function